cli/cmd/generate_gcp.go

```diff
@@ -0,0 +1,780 @@
+						return errors.Wrap(err, "failed to load saved options")
+					}
+				}
+
+				// If the user decides NOT to use the previous values; we won't load them.  However, every time the command runs
+				// we are going to write out new cached values, so if they run it - bail out - and run it again they'll get
+				// re-prompted.
+				if answer {
+					// Merge cached inputs to current options (current options win)
+					if err := mergo.Merge(GenerateGcpCommandState, cachedOptions); err != nil {
+						return errors.Wrap(err, "failed to load saved options")
+					}
+					if err := mergo.Merge(GenerateGcpCommandExtraState, extraState); err != nil {
+						return errors.Wrap(err, "failed to load saved options")
+					}
+				}
+			}
+
+			// Collect and/or confirm parameters
+			err = promptGcpGenerate(GenerateGcpCommandState, GenerateGcpExistingServiceAccountDetails, GenerateGcpCommandExtraState)
+			if err != nil {
+				return errors.Wrap(err, "collecting/confirming parameters")
+			}
+
+			return nil
+		},
+	}
+)
+
+func initGenerateGcpTfCommandFlags() {
+	// add flags to sub commands
+	// TODO Share the help with the interactive generation
+	generateGcpTfCommand.PersistentFlags().BoolVar(
+		&GenerateGcpCommandState.AuditLog,
+		"audit_log",
+		false,
+		"enable audit log integration")
+	generateGcpTfCommand.PersistentFlags().BoolVar(
+		&GenerateGcpCommandState.Config,
+		"config",
+		false,
+		"enable config integration")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.ServiceAccountCredentials,
+		"service_account_credentials",
+		"",
+		"specify a Service Account credentials JSON path (leave blank to make use of google credential ENV vars)")
+	generateGcpTfCommand.PersistentFlags().BoolVar(
+		&GenerateGcpCommandState.OrganizationIntegration,
+		"organization_integration",
+		false,
+		"enable organization integration")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.GcpOrganizationId,
+		"organization_id",
+		"",
+		"specify the organization id (only set if organization_integration is set)")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.GcpProjectId,
+		"project_id",
+		"",
+		"specify the project id to be used to provision lacework resources")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpExistingServiceAccountDetails.Name,
+		"existing_service_account_name",
+		"",
+		"specify existing service account name")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpExistingServiceAccountDetails.PrivateKey,
+		"existing_service_account_private_key",
+		"",
+		"specify existing service account key (base64 encoded)")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.ConfigIntegrationName,
+		"config_integration_name",
+		"",
+		"specify a custom config integration name")
+	// TODO: Implement AuditLogLabels, BucketLabels, PubSubSubscriptionLabels & PubSubTopicLabels
```

Comment:
@rmoles still planning on this in this PR or should we raise an additional issue?

Reply:
Sounds reasonable I am not sure the initial implementation would need all the 'i's dotted and the 't's crossed